config/config.go

```diff
@@ -1372,6 +1372,7 @@ func getGoGCEnv() int {
 // OTLPConfig is the configuration for writing to the OTLP endpoint.
 type OTLPConfig struct {
 	PromoteResourceAttributes []string `yaml:"promote_resource_attributes,omitempty"`
+	TranslateDots             bool     `default:"true" yaml:"translate_dots"`
```

Comment:
as said in the meeting, let's make this an Enum so that there is future-proofing for other possible translation modes.  Something like "NameConversionMode" with values of "NoConversion" and "ClassicPrometheusConverstion"